agents/node/vcxagent-core/demo/alice.js

```diff
@@ -40,57 +39,46 @@ async function runAlice (options) {
 
   await initRustapi(process.env.VCX_LOG_LEVEL || 'vcx=error')
   const agentName = `alice-${uuid.v4()}`
-  const vcxClient = await createVcxAgent({
+  const connectionId = 'alice-to-faber'
+  const holderCredentialId = 'alice-credential'
+  const disclosedProofId = 'alice-proof'
+  const vcxAgent = await createVcxAgent({
     agentName,
     protocolType: options.protocolType,
     agencyUrl: 'http://localhost:8080',
     seed: '000000000000000000000000Trustee1',
     usePostgresWallet: false,
     logger
   })
-  await vcxClient.updateWebhookUrl(`http://localhost:7209/notifications/${agentName}`)
+  await vcxAgent.agentInitVcx()
+  await vcxAgent.updateWebhookUrl(`http://localhost:7209/notifications/${agentName}`)
 
   const invitationString = await getInvitationString(options['autofetch-invitation-url'])
-  const connectionToFaber = await vcxClient.inviteeConnectionAcceptFromInvitation(agentName, invitationString)
+  const connectionToFaber = await vcxAgent.serviceConnections.inviteeConnectionAcceptFromInvitationAndProgress(connectionId, invitationString)
 
   if (!connectionToFaber) {
     throw Error('Connection with alice was not established.')
   }
   logger.info('Connection to alice was Accepted!')
 
-  await vcxClient.waitForCredentialOfferAndAccept({ connectionName: agentName })
+  await vcxAgent.serviceCredHolder.waitForCredentialOfferAndAcceptAndProgress(connectionId, holderCredentialId)
 
-  logger.info('Poll agency for a proof request')
-  let requests = await DisclosedProof.getRequests(connectionToFaber)
-  while (requests.length === 0) {
-    await sleepPromise(2000)
-    requests = await DisclosedProof.getRequests(connectionToFaber)
+  const proofRequest = (await vcxAgent.serviceProver.waitForProofRequests(connectionId))[0]
```

Comment:
As we know already, this is quite dangerous, as the request is selected undeterministically in case multiple proof requests are received - but it should not lead to issues in this test. However, I would add an assert just in case.

Reply:
It's test so I know there will be exactly 1 offer pending.